src/js/services/quiz.service.ts

```diff
@@ -8,17 +8,51 @@ import { Observable } from 'rxjs/Observable';
 import { PlaylistService } from './playlist.service';
 import { QuestionsComponent } from '../components/questions/questions.component';
 
+import { Track } from '../models/track';
+
 @Injectable()
 export class QuizService {
 
   private _questionsComponent: QuestionsComponent;
   private _totalQuestions = 0;
+  private _tracks: Track[];
 
   constructor(private playlistService: PlaylistService) {
     this.playlistService.getPlaylist().subscribe(
       playlist => this._totalQuestions = 5,
       error => console.error(error)
       );
+
+    if (this._totalQuestions > this.playlistService.tracks.length) {
+      this._totalQuestions = this.playlistService.tracks.length;
+    }
+    this._tracks = this.getRandomTracks(this._totalQuestions);
```

Comment:
That's because you have to access the data in the subscribe callback function, otherwise the data will not be available yet. I'm revisiting this in a new branch **services**.